pkg/server/skydb/pq/device.go

```diff
@@ -24,16 +24,46 @@ import (
 )
 
 func (c *conn) GetDevice(id string, device *skydb.Device) error {
-	builder := psql.Select("type", "token", "user_id", "last_registered_at").
+	builder := psql.Select("type", "token", "user_id", "topic", "last_registered_at").
 		From(c.tableName("_device")).
-		Where("id = ?", id)
+		Where("id = ? AND topic IS NOT NULL", id)
+
+	nullableToken := sql.NullString{}
+	nullableUserID := sql.NullString{}
+	err := c.QueryRowWith(builder).Scan(
+		&device.Type,
+		&nullableToken,
+		&nullableUserID,
+		&device.Topic,
+		&device.LastRegisteredAt,
+	)
+
+	if err == sql.ErrNoRows {
+		return skydb.ErrDeviceNotFound
+	} else if err != nil {
+		return err
+	}
+
+	device.Token = nullableToken.String
+	device.UserInfoID = nullableUserID.String
+	device.LastRegisteredAt = device.LastRegisteredAt.In(time.UTC)
+	device.ID = id
+
+	return nil
+}
+
+func (c *conn) GetDeviceByIDAndTopic(id, topic string, device *skydb.Device) error {
```

Comment:
There is zero or one device that will match an ID, therefore the topic is redundant here.

This also pose problems with the caller having to decide whether to call `GetDevice` or `GetDeviceByIDAndTopic`. In golang, you used function pointer, which seems to much work.

I am more inclined if you merge these two functions and adopt one of the following approach:

* if `topic` is empty string, ignore the `topic` filter
* remove the `topic` argument, and the caller is supposed to check whether the return devices has the correct topic (which is the job of the pusher IMO)